src/url-ref.ts

```diff
@@ -1,20 +1,41 @@
-import { ref, watch, Ref } from "vue";
+import { watch, Ref, shallowRef } from "vue";
 import { RouteLocationNormalized, Router } from "vue-router";
 
 export function useUrlRef(router: Router, route: RouteLocationNormalized) {
-  function urlRef(name: string, defaultValue: string): Ref<string> {
-    const value = ref<string>("" + (route.query[name] ?? defaultValue));
+  function urlRef(name: string, defaultValue: boolean): Ref<boolean>;
+  function urlRef(name: string, defaultValue: string): Ref<string>;
+  function urlRef<T extends string | boolean>(name: string, defaultValue: T): Ref<T> {
+    // Parse the value in the URL
+    const type = typeof defaultValue;
```

Comment:
this leads to problems when loading a page without url extensions, try the truth table for example, that now starts with undefined in the input field

Reply:
I implemented a fix, please review if possible